### PR TITLE
CA-217844: Handling Automatic update not supported

### PR DESCRIPTION
**Pull request by GaborApatiNagy**
If there is no <minimalpatches /> tag, the version is not supported. This check has priority over licensed state when displaying the reason why a server is not supported.

Signed-off-by: Gabor Apati-Nagy <gabor.apati-nagy@citrix.com>

### CODE REVIEW

File: XenModel/Messages.resx [hunk 1]
```diff
@@ -9006,6 +9006,9 @@ However, there is not enough space to perform the repartitioning, so the current
   <data name="PATCHINGWIZARD_SELECTPATCHPAGE_UPDATESEXT" xml:space="preserve">
     <value>[XenServer] Updates and Supplemental Packs (*.{0}, *.iso)|*.{0};*.iso</value>
   </data>
+  <data name="PATCHINGWIZARD_SELECTSERVERPAGE_AUTO_UPDATE_NOT_SUPPORTED_HOST_VERSION" xml:space="preserve">
+    <value>Automatic update is not supported on this [XenServer] version.</value>
```
Comment (kc284) [inline XenModel/Messages.resx]: You may want to remove the full stop fromt his tooltip too :)

File: XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs [hunk 2]
```diff
@@ -254,12 +254,14 @@ private void GetXenServerVersions(XmlDocument xdoc)
                     }
 
                     List<XenServerPatch> patches = new List<XenServerPatch>();
```
Comment (kc284) [inline XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs]: Should we be doing the same with patches? What would happen if the patch node were absent?

File: XenAdmin/Core/Updates.cs [hunk 3]
```diff
@@ -442,6 +442,9 @@ public static List<XenServerPatch> RecommendedPatchesForHost(Host host)
                 var minimumPatches = new List<XenServerPatch>();
```
Comment (kc284) [inline XenAdmin/Core/Updates.cs]: These two lines can be compacted:
`var minimumPatches = serverVersions[0].MinimalPatches;`

Comment (GaborApatiNagy) [inline XenAdmin/Core/Updates.cs]: Good catch! Let me not fix this here, I have been already working on my next PR that will fix this.

File: XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs (hunk 2 above)
Comment (GaborApatiNagy) [inline XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs]: Comment I accidentally deleted by @kc284: "Should we be doing the same with patches? What would happen if the patch node were absent?"

Comment (GaborApatiNagy) [inline XenModel/Actions/Updates/DownloadUpdatesXmlAction.cs]: Well, I could, but it wouldn't make any difference to existing functionalities. This is a must for minimalPatches, but not needed for the others.

I think -as part of my bugfix- I am better off not touching what I don't have to and live exactly it as it's been before.

As a future improvement, I'd rather suggest we completely get rid off this code and parse the XML with a DOM parser. (Actually, it has already got some issues like: handling duplicate nodes and not validating the document against a schema)

File: XenModel/Messages.resx (hunk 1 above)
Comment (GaborApatiNagy) [inline XenModel/Messages.resx]: Will do, thanks!

**Comment by GaborApatiNagy** [comment]
Thanks @kc284 for reviewing, please have a look at my most recent commit.

**Comment by kc284** [comment]
Ok, merging.

**Pull request merged**